src/MongoDB.Driver.Core/Core/Authentication/SaslAuthenticator.cs

```diff
@@ -213,7 +213,7 @@ private MongoAuthenticationException CreateException(IConnection connection, Exc
         /// <summary>
         /// Represents a SASL conversation.
         /// </summary>
-        protected sealed class SaslConversation : IDisposable
+        public sealed class SaslConversation : IDisposable
```

Comment:
Does it need to be public now?

Reply:
Registering unmanaged wrappers for disposal was moved from `GssapiAuthenticator` into `SspiSecurityContext`, which required broadening the accessibility of the nested `SaslAuthenticator.SaslConversation` class. I'm not totally happy with how these pieces fit together logically, but it was good enough for now while minimizing potentially breaking changes. As well `SaslAuthenticator` and `SaslAuthenticator.SaslConversation` are used by more than just SSPI. I didn't want to risk breaking LDAP, AWS, and other SASL-based mechanisms while making the SSPI/Kerberos code prettier.

That all said, I agree that making this internal class public increases our API surface area unnecessarily and that `protected internal` is sufficient. Thanks for catching this.